src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4956,6 +5033,101 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
         tmp = asr;
     }
+    
+    void visit_Interactive(const AST::Interactive_t &x) {
+        static size_t interactive_execution_count = 0; // ???: should this be a class member variable
+        // interactive_execution_count should be the same as eval_count in PythonCompiler
```

Comment:
Any suggestions on this?
Should we create `run_fn1` and `run_fn2` and share it through the `PassOptions`?
But even that may not be straightforward.

Reply:
Another option would be to combine the `init` and `stmts` functions. Look at #2573

Reply:
I saw the linked issue. If combining `init` and `stmt` function solves your issue, then you can try it. Submit it as a separate PR and we can rebase this after that is merged.